Add a few more projects

ianthetechie opened the pull request:
Adding a few more here. I have a few other things we should add too, but wanted to start a discussion first on organization.

The "other official projects" designation is quite confusing. For example, it includes MapLibre Navigation Android, but the Qt and Flutter packages are listed under bindings. So it's not particularly clear where the SwiftUI DSL should end up ;) (I also noticed that the iOS Navigation SDK is missing too, but will add that after we resolve the discussion points below.)

A few ideas:

1. If we are OK repeating items, it might make sense to put both these under multiple headings, as they are all (AFAIK) "official" projects, but that is not particularly useful for discovery groupings. I'm also not sure (this is a really subjective view) if "bindings" is even a very helpful heading. Browsers probably care more about "what are ways that I can use MapLibre and <Technology X>." So, for a concrete suggestion, we could have categories for iOS, Qt, and Android. Projects could be listed in multiple categories then for better discoverability. This would also make more logical points where associated/deeply integrated libraries, such as Ferrostar, can fit in (ex: we could have categories for iOS, Android, and Turn-by-Turn Navigation).
2. If we don't want to repeat items, I'm not exactly sure what to do. I personally find the "other official projects" label not super helpful so I'd lean toward breaking up that list and putting items into other categories. In this scenario, it may still be useful though to highlight which projects are "official." I propose we could do that with an emoji or something with an explainer at the top. For example, ⭐? This gets into the second confusion though with what constitutes "official." Maybe we should differentiate between the projects with monetary support and the hosted ones?

Anyways, thanks for reading and I hope the ensuing discussion makes this list more awesome :)

HarelM commented:
TBH, I'll be surprised if anyone is actually reading this, let alone start to finish. 
But I can relate to the need to order stuff 😁
An emoji for core project (those officially maintain by paid maplibre engineers) works work. A different emoji for hosed (under the maplibre org) can be fine too. 
And in general I don't have strong feelings either way. 
Not that part of the information here is copied to the maplibre-gl docs, so let's make sure this doesn't break...

ianthetechie commented:
> TBH, I'll be surprised if anyone is actually reading this, let alone start to finish.

Fair point 😂 

Ok, I just took a swing at reorganizing a few things. Here's what changed at a high level.

# Emoji

I went with ✅ for core projects, and 💙 for hosted, and put a note at the top.

# Category reshuffle

This frees up a bit better classification of projects by what they do / why a reader might care. Obviously the "official map rendering" and "other official" categories went away. I kept map rendering at the top.

We have a TON of projects that are all pretty closely related to "map styling" for lack of a better term. I put those in a hierarchy with a few sub-sections and rearranged the projects to match. This cleared out about half of the "utilities" section which wasn't very helpful as a classifier IMO.

We also have a lot of navigation/directions projects (including an official project that wasn't listed 😬), so I fixed that.

I had a very bad time coming up with buckets for all the mobile stuff. Bindings evokes the wrong image as a category, but most other stuff isn't really what I think of as bindings either, so I rolled with it. I put all of the higher-level mobile SDKs under Bindings > Mobile, and expanded the category with the developments from the past ~year.

Nothing in the gigantic middle really changed, except that I moved MapLibre GL Leaflet to a logical place in the JS hierarchy (verified the docs generator code and it should be fine as far as I can tell; looks like it parses comments with a regex).

Couldn't find a good spot for the demo tiles, so I put them under map/tile providers 🤷 

And finally, map tile generators also emerged as a natural category.

HarelM commented on README.md:
```diff
@@ -196,21 +222,17 @@ A collection of awesome things that use or support [MapLibre](https://maplibre.o
 **In bold**: Members of the [MapLibre Sponsorship Program](https://maplibre.org/sponsors/)
 
-## Utilities
+## Map Tile Generators
```
Tile Serving? Tile Servers?
There's also tileserver-gl fork that migrated to use maplibre native, I think.It might even be a hosted project, IDK

ianthetechie replied:
Oh, yes! I'm surprised TileServer GL wasn't in here already! Can't find a fork, but the original is a MapTiler project that's using MapLibre Native so I'll add that.

HarelM commented:
I've just read it start to finish 🤪
Looks good. 
I would consider moving the hosted projects to be at the top of their category, but this is mainly nitpicking...

HarelM commented on README.md:
```diff
@@ -4,31 +4,54 @@
+## Bindings
 
-- [flutter-maplibre-gl](https://github.com/maplibre/flutter-maplibre-gl) - Bindings for Flutter on Android, iOS and Web, on pub.dev at [pub.dev/packages/maplibre_gl](https://pub.dev/packages/maplibre_gl)
+###  Mobile
```
Other header are for languages, this is a bit "off", consider splitting this to swift, react-native, flutter, etc...

ianthetechie replied:
Yeah... I thought about that... The reason I coalesced them together is that it felt a bit silly with a bunch of single-item categories. I guess maybe it can't be helped, and the suggestion is indeed better for skimmability.

ianthetechie commented:
> I've just read it start to finish 🤪

ha! Thank you for your patience 😂 

> I would consider moving the hosted projects to be at the top of their category, but this is mainly nitpicking...

Good call.
